src/containers/PatientDetails/PatientDocument/index.tsx

```diff
@@ -23,16 +28,99 @@ export interface PatientDocumentProps {
     questionnaireId?: string;
     encounterId?: string;
     onSuccess?: (resource: QuestionnaireResponseFormSaveResponse) => void;
-    autosave?: boolean;
+    autoSave?: boolean;
     qrDraftServiceType?: QuestionnaireResponseDraftService;
 }
 
+interface PatientDocumentContentProps extends PatientDocumentProps {
+    onCancel?: () => void;
+    onQRFUpdate?: (questionnaireResponse: QuestionnaireResponse) => void;
+    alertComponent?: React.ReactNode | (() => React.ReactNode);
+}
+
 export function PatientDocument(props: PatientDocumentProps) {
-    const { autosave, qrDraftServiceType = 'local' } = props;
+    const { autoSave = false, qrDraftServiceType = 'local' } = props;
+
+    const params = useParams<{ questionnaireId: string; encounterId?: string }>();
+
+    const {
+        response,
+        draftInfoMessage,
+        updateDraft: onUpdateDraft,
+        deleteDraft,
+        autoSaveEnabled,
+        setAutoSaveEnabled,
+    } = useQuestionnaireResponseDraft({
+        subject: `${props.patient.resourceType}/${props.patient.id}`,
+        questionnaireId: props.questionnaireId ?? params.questionnaireId!,
+        qrDraftServiceType,
+        autoSave,
+        questionnaireResponse: props.questionnaireResponse,
+    });
+
+    return (
+        <RenderRemoteData remoteData={response} renderLoading={() => <div>Hello</div>}>
+            {(draftQuestionnaireResponse) => (
+                <PatientDocumentContent
+                    {...props}
+                    key="patient-document-content"
+                    questionnaireResponse={draftQuestionnaireResponse}
+                    onSuccess={async (resource: QuestionnaireResponseFormSaveResponse) => {
+                        await deleteDraft();
+                        props.onSuccess && props.onSuccess(resource);
+                    }}
+                    onCancel={deleteDraft}
+                    onQRFUpdate={onUpdateDraft}
+                    alertComponent={
+                        <AlertMessage
+                            actionComponent={
+                                <Space>
+                                    <Tooltip title={t`Enable or disable auto saving to ${qrDraftServiceType} storage`}>
```

Comment:
Personally, I don't like how it looks - too many controls. And I don't understand the use case of turning off autosave for local.
It makes sense for server, but totally does not make sense for local.
And the clear button - it makes sense only for local.
And I prefer to get rid of "x" button at all - it's super confusing.
And the clear icon is not obvious, maybe it will be better to use thrash icon, but let's ask designer.

<img width="667" height="84" alt="Screenshot 2025-07-29 at 22 13 30" src="https://github.com/user-attachments/assets/21d10375-e395-411e-b632-65e36c01a1e3" />